script/dev/dev-UpgradeLockingContract.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.18;
+
+import { console2 } from "forge-std/Script.sol";
+import { Script } from "script/utils/mento/Script.sol";
+import { Chain as ChainLib } from "script/utils/mento/Chain.sol";
+import { Contracts } from "script/utils/mento/Contracts.sol";
+import { IRegistry } from "script/interfaces/IRegistry.sol";
+import { Locking } from "contracts/locking/Locking.sol";
+import { IGovernanceFactory } from "script/interfaces/IGovernanceFactory.sol";
+import { IERC20Upgradeable } from "openzeppelin-contracts-upgradeable/contracts/token/ERC20/IERC20Upgradeable.sol";
+
+/**
+ * Usage: yarn script:dev -n alfajores -s UpgradeLockingContract -r "run()"
+ * Used to deploy the Locking V2 implementation
+ * ===========================================================
+ */
+contract UpgradeLockingContract is Script {
+  using Contracts for Contracts.Cache;
+
+  function run() public {
+    contracts.load("MUGOV-00-Create-Factory", "latest");
+    address lockingV2;
+    address mentoToken = IGovernanceFactory(contracts.deployed("GovernanceFactory")).mentoToken();
+    vm.startBroadcast(ChainLib.deployerPrivateKey());
+    {
+      lockingV2 = address(new Locking());
+      console2.log("----------");
+      console2.log("LockingV2 deployed at: ", lockingV2);
+      console2.log("----------");
+      Locking(lockingV2).__Locking_init(IERC20Upgradeable(mentoToken), 212, 0, 1);
```

Comment:
lets maybe move these parameters into variables and add a comment for context to specify they are taken from mainnet?

Reply:
Roger that